eyeball-im-util/src/vector/ops.rs

```diff
@@ -1,71 +1,75 @@
-use std::collections::VecDeque;
-
+use arrayvec::ArrayVec;
 use eyeball_im::VectorDiff;
 
-pub trait VectorDiffContainerOps<T> {
+pub trait VectorDiffContainerOps<T>: Sized {
     type Family: VectorDiffContainerFamily;
+    type Buffer: Default;
 
     fn from_item(vector_diff: VectorDiff<T>) -> Self;
 
-    fn pick_item(vector_diffs: &mut VecDeque<Self>) -> Option<Self>
-    where
-        Self: Sized;
-
-    fn for_each(self, f: impl FnMut(VectorDiff<T>));
-
     fn filter_map<U>(
         self,
         f: impl FnMut(VectorDiff<T>) -> Option<VectorDiff<U>>,
     ) -> Option<VectorDiffContainerFamilyMember<Self::Family, U>>;
+
+    fn push_into_buffer(
+        self,
+        buffer: &mut Self::Buffer,
+        make_diffs: impl FnMut(VectorDiff<T>) -> ArrayVec<VectorDiff<T>, 2>,
```

Comment:
Should we parameter the size of `ArrayVec` instead of statically use 2 here? That's probably a premature optimisation though.

Reply:
Good to me.